src/stream.cpp

```diff
@@ -122,8 +124,10 @@ namespace oxen::quic
 
     void Stream::when_available(unblocked_callback_t unblocked_cb)
     {
-        log::trace(log_cat, "{} called", __PRETTY_FUNCTION__);
-        unblocked_callbacks.push(std::move(unblocked_cb));
+        conn.quic_manager->call([this, unblocked_cb]() {
+            log::trace(log_cat, "{} called", __PRETTY_FUNCTION__);
+            unblocked_callbacks.push(std::move(unblocked_cb));
```

Comment:
There is some implicit constness here that is mucking things up (not worth changing because this function is going away) but:
- lambda value captures are const by default (declare the lambda `mutable` to fix that)
- because it's const, the `std::move(unblocked_cb)` is a `const&&` rather than a `&&` rvalue reference, and that *can't* bind to the usual rvalue move operator, so will have to do a copy (and so the `std::move` does nothing, but is subtly misleading).

Reply:
Also: the `unblocked_cb` capture should be `unblocked_cb=std::move(unblocked_cb)` to more it into the lambda captures.